boot/fdehook/fdehook.go

```diff
@@ -0,0 +1,147 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package fdehook
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/snapcore/snapd/osutil"
+)
+
+func fdeHook(kernelDir string) string {
+	return filepath.Join(kernelDir, "meta/hooks/fde")
```

Comment:
what's the plan here, to have  two copies of the hook inside the kernel snap, one out the efi under meta? and one inside the initrd but at with some of the same parent tree?

Reply:
That's a really good question. I think we will probably have to live with two copies. Alternatively we could split the helper into a "lock" and "unlock" helper. The later option might be clearer and it will also make very clear what part should be in the initrd and what part in the hooks dir. Wdyt?

Reply:
also to be clear I don't think it should go to hooks if it's not called like the other hooks

Reply:
I renamed fdehook -> fdehelper as this is not a hook in the snapd sense.